src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -186,108 +205,165 @@ internal void UpdateSelectedItem(PackageItemListViewModel selectedItem)
-                    loadCts.Dispose();
-                    currentLoader.Reset();
+                    // Ideally, After a search, it should report its status and,
+                    // do not keep the LoadingStatus.Loading forever.
+                    // This is a workaround.
+                    var emptyListCount = addedLoadingIndicator ? 1 : 0;
+                    if (Items.Count == emptyListCount)
+                    {
+                        _loadingStatusIndicator.Status = LoadingStatus.NoItemsFound;
+                    }
+                    else
+                    {
+                        Items.Remove(_loadingStatusIndicator);
+                    }
+                }
+            }
 
-                    // Write stack to activity log
-                    Mvs.ActivityLog.LogError(LogEntrySource, ex.ToString());
+            UpdateCheckBoxStatus();
 
-                    await _joinableTaskFactory.Value.SwitchToMainThreadAsync();
+            LoadItemsCompleted?.Invoke(this, EventArgs.Empty);
+        }
 
-                    var errorMessage = ExceptionUtilities.DisplayMessage(ex);
-                    _logger.Log(ProjectManagement.MessageLevel.Error, errorMessage);
+        internal void FilterItems(ItemFilter itemFilter, CancellationToken token)
+        {
+            if (!Items.Contains(_loadingStatusIndicator))
+            {
+                Items.Add(_loadingStatusIndicator);
+            }
+            _loadingStatusIndicator.Status = LoadingStatus.Loading;
 
-                    _loadingStatusIndicator.SetError(errorMessage);
+            // If there is another async loading process - cancel it.
+            var loadCts = CancellationTokenSource.CreateLinkedTokenSource(token);
+            Interlocked.Exchange(ref _loadCts, loadCts)?.Cancel();
 
-                    _loadingStatusBar.SetError();
-                    _loadingStatusBar.Visibility = Visibility.Visible;
+            try
+            {
+                if (itemFilter == ItemFilter.UpdatesAvailable)
+                {
+                    ApplyItemsFilterForUpdatesAvailable();
+                }
+                else
+                {
+                    //Show all the items, without an Update filter.
+                    ClearItemsFilter();
+                }
+            }
+            finally
+            {
+                //If no items are shown in the filter, indicate in the list that no packages are found.
+                if (FilterCount == 0)
+                {
+                    _loadingStatusIndicator.Status = LoadingStatus.NoItemsFound;
                 }
-                finally
+                else
                 {
-                    if (_loadingStatusIndicator.Status != LoadingStatus.NoItemsFound
-                        && _loadingStatusIndicator.Status != LoadingStatus.ErrorOccurred)
+                    //There are packages, but since no loading will occur, there's no need for an indicator.
```

Comment:
perhaps clarify comment a bit. i know, because of your PR description, that no loading will occur because there is no pagination enabled anymore in updates or installed tabs.
Perhaps:
// Installed and Updates tab don't need an indicator shown in the case that FilteredItems exist...since there is no pagination, and thus the indicator doesn't need to display "loading"

Reply:
There is no change in behavior from what we had, so I'll just remove my comment entirely.